dora/core/server/worker/src/main/java/alluxio/worker/modules/LegacyWorkerIdentityProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.modules;
+
+import alluxio.exception.runtime.AlluxioRuntimeException;
+import alluxio.resource.PooledResource;
+import alluxio.retry.RetryPolicy;
+import alluxio.retry.RetryUtils;
+import alluxio.wire.WorkerIdentity;
+import alluxio.worker.WorkerProcess;
+import alluxio.worker.block.BlockMasterClient;
+import alluxio.worker.block.BlockMasterClientPool;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Provider that provides worker identity by registering with the master.
+ */
+public class LegacyWorkerIdentityProvider implements Provider<WorkerIdentity> {
+  private static final Logger LOG = LoggerFactory.getLogger(LegacyWorkerIdentityProvider.class);
+
+  private final BlockMasterClientPool mBlockMasterClientPool;
+  private final WorkerProcess mWorkerProcess;
+
+  /**
+   * Constructor.
+   *
+   * @param bmcPool
+   * @param workerProcess
+   */
+  @Inject
+  public LegacyWorkerIdentityProvider(BlockMasterClientPool bmcPool, WorkerProcess workerProcess) {
```

Comment:
where is this used? i didnt see it used in pageddoraworker registertomaster either? can u remove this if its not used? we dont want to write code only to serve the purpose of getting deprecated in the future

Reply:
Yes, this should be removed. It was a failed attempt to extract the worker registration process from `PagedDoraWorker` as a guice provider.